Gems/Atom/RPI/Code/Include/Atom/RPI.Reflect/Material/MaterialAsset.h

```diff
@@ -105,6 +105,15 @@ namespace AZ
             //! their parent material, they all get flattened at build time so every MaterialAsset has the full set of values.
             AZStd::array_view<MaterialPropertyValue> GetPropertyValues() const;
 
+            bool HasPropertyNames() const { return !m_propertyNames.empty(); }
+            const AZStd::vector<AZ::Name>& GetPropertyNames() const { return m_propertyNames; }
+
+            // temp code
+            const MaterialPropertyValue& GetDefaultPropertyValue(size_t index) const
+            {
+                return m_materialTypeAsset->GetDefaultPropertyValues()[index];
+            }
+
```

Comment:
I would recommend expanding all of these and moving them to the source file. Possibly also add comments.

Reply:
Thanks for looking! this was just a temp PR, the real one is over here: https://github.com/o3de/o3de/pull/3408